examples/with-stripe-typescript/app/components/CheckoutForm.tsx

```diff
@@ -0,0 +1,48 @@
+'use client'
+
+import React, { useState } from 'react'
+
+import CustomDonationInput from '../../components/CustomDonationInput'
+import StripeTestCards from '../../components/StripeTestCards'
+
+import { formatAmountForDisplay } from '../../utils/stripe-helpers'
+import * as config from '../../config'
+import { createCheckoutSession } from '../actions/stripe'
+
+const CheckoutForm = () => {
+  const [loading, setLoading] = useState(false)
+  const [input, setInput] = useState({
+    customDonation: Math.round(config.MAX_AMOUNT / config.AMOUNT_STEP)
+  })
+
+  const handleInputChange: React.ChangeEventHandler<HTMLInputElement> = (e) =>
+    setInput({
+      ...input,
+      [e.currentTarget.name]: e.currentTarget.value
+    })
+
+  return (
+    <form action={createCheckoutSession}>
```

Comment:
Does this need to be a client component?

Reply:
@leerob The `'use client'` directive is currently required because of `useState` hook usage (for handling some UI state). I can refactor if you think that's important.